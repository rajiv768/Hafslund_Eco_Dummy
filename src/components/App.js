import 'core-js/stable';
import 'regenerator-runtime/runtime';

import React from 'react';
import PropTypes from 'prop-types';

import ErrorBoundary from './ErrorBoundary';
import { WidgetProvider } from './Context';
import MyComponent from './MyComponent';

const App = props => {
  return (
    <ErrorBoundary>
      <WidgetProvider
        wab={props.wab}
        esriJS={props.esriJS}
        onOpen={props.onOpen}
        onClose={props.onClose}
      >
        <MyComponent title="WAB Widget" />
      </WidgetProvider>
    </ErrorBoundary>
  );
};

App.propTypes = {
  wab: PropTypes.object,
  esriJS: PropTypes.object,
  onOpen: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
};

App.defaultProps = {
  wab: {},
  esriJS: {},
};

export default App;
