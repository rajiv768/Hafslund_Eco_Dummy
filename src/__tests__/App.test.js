import React from 'react';
import { render, wait, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import App from '../components/App';

import mock from './__mock__/propsMock';

afterEach(cleanup);

// https://testing-library.com/docs/react-testing-library/example-intro
describe('<App />', () => {
  it('renders the widget', async () => {
    // render component
    const { container, getByText, queryByText } = render(
      <App
        wab={mock.wab}
        esriJS={mock.esriJS}
        onOpen={jest.fn}
        onClose={jest.fn}
      />,
    );

    // test inital render
    expect(getByText('Loading...')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();

    // wait for useEffect
    await wait(() =>
      expect(queryByText(/Loading\.\.\./i)).not.toBeInTheDocument(),
    );

    // this will be error bc we cant access the JS api from the test
    // but can test post useEffect
    expect(getByText('Error...')).toBeInTheDocument();
  });
});
