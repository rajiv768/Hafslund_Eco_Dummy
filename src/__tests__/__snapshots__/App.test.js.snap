// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the widget 1`] = `
<div>
  Loading...
</div>
`;
